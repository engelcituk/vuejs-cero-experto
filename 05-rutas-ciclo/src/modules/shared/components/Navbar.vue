<template>
    <div>
        <!-- <router-link :to="{name:'home' }">Pokemon List</router-link>
        <router-link :to="{name:'pokemon-id', params:{id: 34} }">Pokemon por id</router-link>
        <router-link :to="{name:'about'}">About</router-link> -->
        <CustomLink v-for="link in links" :key="link.to" :link="link"/>
    </div>
</template>

<script>
import { defineAsyncComponent } from 'vue'

export default {
    data(){
        return {
            links: [
                { to:'/home', name:'pokemons' },
                { to:'/pokemonid/50', name:'por id' },
                { to:'/about', name:'About' },
                { to:'https://google.com', name:'Google' },

            ]
        }
    },
    components:{
        CustomLink: defineAsyncComponent( () => import('./CustomLink.vue'))
    }    
}
</script>


<style scoped>
    div{
        padding: 0 30px;
    }

    div a{
        font-weight: bold;
        color: #2c3e50;
        margin: 0 10px;
        text-decoration: none;
    }

    /* router-link-active */
    /* div .router-link-exact-active{
        color: #42b983;
    } */
</style>
