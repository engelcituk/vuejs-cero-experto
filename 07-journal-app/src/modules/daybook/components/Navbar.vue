<template>
    <div>
        <nav class="navbar bg-primary">
            <a href="" class="navbar-brand text-white">
                <img src="@/assets/logo.png" alt="vue logo" class="d-inline-block align-text-top mx-2" height="24">
                {{username}}
            </a>
            <div class="d-flex">
                <button class="btn btn-outline-info mx-2" @click="onLogout">
                    <i class="fa fa-sign-out-alt"></i>
                </button>
            </div>
        </nav>
    </div>
</template>

<script>
    import useAuth from '@/modules/auth/composables/useAuth'
    import { useRouter } from 'vue-router'
    export default {
        setup(){
            const router = useRouter()

            const { username, logout } = useAuth()

            return {
                username,
                onLogout: () => {
                    logout()
                    router.push( {name: 'login' } )
                }
            }
        }
    }
</script>

<style scoped>

</style>