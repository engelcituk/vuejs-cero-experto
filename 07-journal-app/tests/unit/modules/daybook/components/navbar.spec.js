import { shallowMount } from '@vue/test-utils'
import Navbar from '@/modules/daybook/components/Navbar'
import createVuexStore from '../../../mock-data/mock-store'


// para correr la prueba: npm run test:unit navbar
describe('Pruebas en el componente navBar', () => {
    const store = createVuexStore({
            user :{
                name: 'Juan Carlos',
                email: 'juam@mail.com'
            },
            status: 'authenticated',
            idtoken: 'ABC-123',
            refreshToken: 'XYZ'
    })
    
    test('Debe de mostrar el componente correctament', () => { 
        const wrapper = shallowMount(Navbar, {
            global:{
                plugins:[ store ]
            }
        })
        expect( wrapper.html() ).toMatchSnapshot()
    })
   
})