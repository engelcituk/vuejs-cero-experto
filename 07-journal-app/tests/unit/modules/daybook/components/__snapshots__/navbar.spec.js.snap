// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente navBar Debe de mostrar el componente correctament 1`] = `
<div>
  <nav class="navbar bg-primary"><a href="" class="navbar-brand text-white"><img src="" alt="vue logo" class="d-inline-block align-text-top mx-2" height="24"> Juan Carlos</a>
    <div class="d-flex"><button class="btn btn-outline-info mx-2"><i class="fa fa-sign-out-alt"></i></button></div>
  </nav>
</div>
`;
